pkg/cedar/translated_client.go

```diff
@@ -50,3 +56,128 @@ func (c TranslatedClient) FetchSystems(logger *zap.Logger) (models.SystemShorts,
 	}
 	return systems, nil
 }
+
+func validateSystemIntakeForCedar(intake *models.SystemIntake, logger *zap.Logger) error {
+	var res []string
+	if validate.RequireUUID(intake.ID) {
+		res = append(res, "ID")
+	}
+	if validate.RequireString(intake.EUAUserID) {
+		res = append(res, "EUAUserID")
+	}
+	if validate.RequireNullString(intake.Requester) {
+		res = append(res, "Requester")
+	}
+	if validate.RequireNullString(intake.Component) {
+		res = append(res, "Component")
+	}
+	if validate.RequireNullString(intake.BusinessOwner) {
+		res = append(res, "BusinessOwner")
+	}
+	if validate.RequireNullString(intake.BusinessOwnerComponent) {
+		res = append(res, "BusinessOwnerComponent")
+	}
+	if validate.RequireNullString(intake.ProductManager) {
+		res = append(res, "ProductManager")
+	}
+	if validate.RequireNullString(intake.ProductManagerComponent) {
+		res = append(res, "ProductManagerComponent")
+	}
+	if validate.RequireNullString(intake.ProjectName) {
+		res = append(res, "ProjectName")
+	}
+	if validate.RequireNullBool(intake.ExistingFunding) {
+		res = append(res, "ExistingFunding")
+	}
+	if intake.ExistingFunding.Bool && validate.RequireNullString(intake.FundingSource) {
+		res = append(res, "FundingSource")
+	}
+	if validate.RequireNullString(intake.BusinessNeed) {
+		res = append(res, "BusinessNeed")
+	}
+	if validate.RequireNullString(intake.Solution) {
+		res = append(res, "Solution")
+	}
+	if validate.RequireNullBool(intake.EASupportRequest) {
+		res = append(res, "EASupportRequest")
+	}
+	if validate.RequireNullString(intake.ProcessStatus) {
+		res = append(res, "ProcessStatus")
+	}
+	if validate.RequireNullString(intake.ExistingContract) {
+		res = append(res, "ExistingContract")
+	}
+	if validate.RequireTime(*intake.SubmittedAt) {
+		res = append(res, "SubmittedAt")
+	}
+	if len(res) != 0 {
+		invalidFields := strings.Join(res, ", ")
+		return errors.New("Validation failed on these fields: " + invalidFields)
+	}
+	return nil
+}
+
+func submitSystemIntake(validatedIntake *models.SystemIntake, c TranslatedClient, logger *zap.Logger) (string, error) {
```

Comment:
`validatedIntake` can probably b `intake`. Whether it is validated or not is outside of the context of this function (ie. creating an assumption)

Reply:
I wanted to explicitly call this validated so it doesn't get reuse without validating the intake before calling this. I figured this would force us to think about what we pass here more. The validate system intake method can be exported, but this one shouldn't be.